src/Gaffer/Monitor.cpp

```diff
@@ -93,7 +101,16 @@ Monitor::Scope::Scope( const MonitorSet &monitors, bool active )
 			m_monitors.erase( m );
 		}
 	}
+
+	// TODO - the amount of duplication here is pretty ugly in general
```

Comment:
Agreed. Could we just delegate everything to a private constructor that is passed what to put in `m_monitors`?